### DISCUSSION TITLE
Julia kernel not found (or how to configure the path to Julia)

### DISCUSSION THREAD

**Discussion by roland-KA** [quarto-dev/quarto-cli#4219] [Q&A] [answered]
I've just installed Quarto and the Quarto-VS-plugin and wanted to render the example document `walkthrough.qmd`.

Rendering didn't succeed, because Quarto cannot find the Julia kernel. It reports the following error:

```
Starting julia-1.6 kernel...ERROR: 

[Errno 2] No such file or directory: '/Applications/Julia-1.6.app/Contents/Resources/julia/bin/julia'
```

And yes, Quarto is right, on that path there is no Julia installation. I'm using `juliaup` to manage my Julia installations. Therefore Julia is installed in `/opt/homebrew/Cellar/juliaup/1.8.16/bin`. 

So how can I tell Quarto to look there for Julia?

And why does Quarto look for a 1.6-version of Julia? Is it capable of using the current 1.8.x-version?

**Comment by mcanouil** [comment]
Quarto is looking for Jupyter in `PATH`, then it is up to Jupyter to find your Julia install.
Did you follow <https://quarto.org/docs/computations/julia.html>?

What's the output of `quarto check jupyter`?

**Comment by roland-KA** [comment, marked as answer]
The output of `quarto check jupyter` is:

```
[✓] Checking Python 3 installation....OK
      Version: 3.9.7 (Conda)
      Path: /Users/roland/opt/anaconda3/bin/python
      Jupyter: 4.8.1
      Kernels: julia-1.6, python3

[✓] Checking Jupyter engine render....OK
```

After installing `IJulia` in my 1.8 default environment, it seems also to see the 1.8-kernel:

```
[✓] Checking Python 3 installation....OK
      Version: 3.9.7 (Conda)
      Path: /Users/roland/opt/anaconda3/bin/python
      Jupyter: 4.8.1
      Kernels: julia-1.6, julia-1.8, python3

[✓] Checking Jupyter engine render....OK
```

But Quarto still cannot find Julia.

I've also had a look at my PATH-variable. It has `/Applications/Julia-1.6.app/Contents/Resources/julia/bin` included, where Quarto is looking for Julia, but doesn't find it because it's not there. 

So I deleted that path from PATH. But Quarto still tries to look there for Julia.

**Reply by mcanouil** [reply]
Did you specify the "right" kernel and explicitly using `jupyter: julia-1.8` in your yaml header?

**Comment by roland-KA** [comment]
Ha, that's it! 😊👍 ... now it works. Thanks a lot!
